src/core/meshcop/joiner_router.cpp

```diff
@@ -141,15 +138,21 @@ ThreadError JoinerRouter::GetJoinerPort(uint16_t &aJoinerPort)
 
         if (tlv->GetType() == Tlv::kJoinerUdpPort)
         {
-            aJoinerPort = reinterpret_cast<JoinerUdpPortTlv *>(tlv)->GetUdpPort();
-            ExitNow(error = kThreadError_None);
+            mJoinerUdpPort = reinterpret_cast<JoinerUdpPortTlv *>(tlv)->GetUdpPort();
```

Comment:
The Joiner Router can always configure its own Joiner UDP Port and override whatever is in the Commissioner Dataset.  In other words, the Joiner Router should only use the Commissioner Dataset value if:

1. The Commissioner Dataset has a Joiner UDP Port TLV and,
2. The Joiner Router does not have its own Joiner UDP port configured.

Does that make sense?